app/src/auth.py

```diff
@@ -1,100 +1,89 @@
     if user.failed_signin_attempts <= 5:
-        login_user(user, remember=remember) # log the user in and create the user session
+        login_user(user, remember=remember)
         user.failed_signin_attempts = 0
         db.session.commit()
         flash('Successfully logged in! Redirecting to dashboard...')
         return redirect(url_for('main.profile'))
     else:
         flash('This account is locked. Please contact support to unlock your account and reset your password.')
         return redirect(url_for('auth.login'))
-    
 
 @auth.route("/register", methods=["GET", "POST"])
 def register():
     """Register user"""
 
     if request.method == "POST":
 
-        #Access the data submitted to form
         email = request.form.get("email")
         username = request.form.get("username")
         password = request.form.get("password")
         confirm_password = request.form.get("confirm_password")
-        first_name = request.form.get("first_name")
-        last_name = request.form.get("last_name")
+        name = request.form.get("name")
 
-        #Check if the inputs are valid
         user_email = User.query.filter_by(email=email).first()
         if user_email:
             flash("This email already exists.")
-            return redirect(url_for("register"))
+            return redirect(url_for("auth.register"))
         user_name = User.query.filter_by(username=username).first()
         if user_name:
             flash("This username already exists.")
-            return redirect(url_for("register"))
+            return redirect(url_for("auth.register"))
         if len(username) == 0:
             flash("You must provide a username.")
-            return redirect(url_for("register"))
+            return redirect(url_for("auth.register"))
         if len(password) == 0:
             flash("You must provide a password.")
-            return redirect(url_for("register"))
+            return redirect(url_for("auth.register"))
         if confirm_password != password:
-            flash("The passwords are not matching!")
-            return redirect(url_for("register"))
-        if len(first_name) == 0:
-            flash("You must provide your first name.")
-            return redirect(url_for("register"))
-        if len(last_name) == 0:
-            flash("You must provide your last name.")
-            return redirect(url_for("register"))
+            flash("The passwords do not match!")
+            return redirect(url_for("auth.register"))
+        if len(name) == 0:
+            flash("You must provide your name.")
+            return redirect(url_for("auth.register"))
 
-        #Add the user to database and login
-        new_user = User(email=email, username=username, password=new_user.set_password(password), first_name=first_name, last_name=last_name)
+        new_user = User(email=email, username=username, password=password, name=name)
```

Comment:
This line should be: 
new_user = User(email=email, username=username, name=name)
The next line should call the set_password method on this new instance of the User model:
new_user.set_password(password)